mitiq/lre/lre.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Extrapolation methods for Layerwise Richardson Extrapolation (LRE)"""
+
+from functools import wraps
+from typing import Any, Callable, Optional, Union
+
+import numpy as np
+from cirq import Circuit
+
+from mitiq import QuantumResult
+from mitiq.lre import (
+    multivariate_layer_scaling,
+    multivariate_richardson_coefficients,
+)
+from mitiq.zne.scaling import fold_gates_at_random
+
+
+def execute_with_lre(
+    input_circuit: Circuit,
+    executor: Callable[[Circuit], QuantumResult],
+    degree: int,
+    fold_multiplier: int,
+    folding_method: Callable[
+        [Union[Any], float], Union[Any]
+    ] = fold_gates_at_random,
```

Comment:
Let's follow what happens in ZNE here for consistency.

https://github.com/unitaryfund/mitiq/blob/6f912f63f087ac1213290819d85cf16695da5b34/mitiq/zne/zne.py#L70

Reply:
Sorry, I'm missing how `num_to_average` is related here. I'm just suggesting we typehint the folding method differently.

Reply:
That's weird. I was viewing this suggestion locally in VS code. I interpreted 'follow what happens in ZNE for consistency' as use the typehint as well as the additional param. 

![image](https://github.com/user-attachments/assets/ce238189-2aab-40a1-83ad-028c89ca4576)

Reply:
I hadn't modified anything in ZNE which is why it's weird that when you linked L70 in `zne.py`, L71 also showed up. 

The location of the suggestion was not changed.